cmd/image-builder/repos.go

```diff
@@ -16,13 +20,74 @@ var defaultDataDirs = []string{
 	"/usr/share/image-builder/repositories",
 }
 
-var newRepoRegistry = func(dataDir string) (*reporegistry.RepoRegistry, error) {
+type repoConfig struct {
+	DataDir    string
+	ExtraRepos []string
+}
+
+func parseExtraRepo(extraRepo string) ([]rpmmd.RepoConfig, error) {
+	// XXX: we want to eventually support more repos here:
+	// - copr:@osbuild/osbuild (with full gpg retrival via the API)
+	// - https://example.com/repo/fedora-42-x86_64/ (with sslverify=True)
+	// But for now we only support local dirs
+	stat, err := os.Stat(extraRepo)
+	if err != nil {
+		return nil, fmt.Errorf("cannot add extra repository: %v", err)
+	}
+	if !stat.IsDir() {
+		return nil, fmt.Errorf("cannot add extra repository %v: must be a local directory", extraRepo)
+	}
+	extraRepo, err = filepath.Abs(extraRepo)
+	if err != nil {
+		return nil, err
+	}
+
+	checkGPG := false
+	return []rpmmd.RepoConfig{
+		{
+			Id:           extraRepo,
+			Name:         filepath.Base(extraRepo),
+			BaseURLs:     []string{fmt.Sprintf("file:%s", extraRepo)},
```

Comment:
Does this really work? AFAIK, the local paths must be prefixed with `file://` :thinking:

Reply:
Thanks for doing the research. Maybe let's stick to `file:///path/to/repo` in our help and docs.

Reply:
Sure, updated.